### ISSUE TITLE
Updater Progress bar crash with Classic Visual Theme

### ISSUE THREAD

**Issue by Belphemur** [Belphemur/SoundSwitch#194]
[SoundSwitch-crashlog-31_5_2017.zip](https://github.com/Belphemur/SoundSwitch/files/1041339/SoundSwitch-crashlog-31_5_2017.zip)
```
Message: Visual Styles-related operation resulted in an error because no visual style is currently active.
Source: System.Windows.Forms
StackTrace:
   at System.Windows.Forms.VisualStyles.VisualStyleRenderer.IsCombinationDefined(String className, Int32 part)
   at System.Windows.Forms.VisualStyles.VisualStyleRenderer..ctor(String className, Int32 part, Int32 state)
   at System.Windows.Forms.ProgressBarRenderer.InitializeRenderer(VisualStyleElement element)
   at System.Windows.Forms.ProgressBarRenderer.DrawHorizontalBar(Graphics g, Rectangle bounds)
   at SoundSwitch.UI.UserControls.TextProgressBar.OnPaint(PaintEventArgs e) in D:\VS\SoundSwitch\SoundSwitch.UI.UserControls\TextProgressBar.cs:line 48
   at System.Windows.Forms.Control.PaintWithErrorHandling(PaintEventArgs e, Int16 layer)
   at System.Windows.Forms.Control.WmPaint(Message& m)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.DebuggableCallback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

**Comment by Belphemur** [comment]
Sorry, done it fast this morning.

It's linked to using the Classic Theme on Windows 7. 
The updater window crash (the progress bar on it to be exact). I had that problem before with another form.

The problem is linked directly with the fact there is no visual style when the classic mode is activated:

https://msdn.microsoft.com/en-us/library/system.windows.forms.progressbarrenderer(v=vs.110).aspx

The fix would be to check IsSupported: https://msdn.microsoft.com/en-us/library/system.windows.forms.progressbarrenderer.issupported%28v=vs.110%29.aspx?f=255&MSPPError=-2147217396

In the case it's not supported, just update the text and not the progress bar.

**Comment by FireEmerald** [comment]
The problem can't occur on Windows 10 Anniversary Update due to the fact that it seems like they removed the classic theme - i will need a virtual machine to reproduce this crash.

**Comment by Belphemur** [comment]
What annoys me is to know how old this issue really is ... But well, I should have checked the documentation before using this progress bar.

I added this progress bar in the beginning of the version 3.x.

**Comment by FireEmerald** [comment]
![2017-05-31 16_26_11](https://cloud.githubusercontent.com/assets/1107939/26636745/f81c61c4-461d-11e7-806e-851cdf35075e.png)
Think the worse bug is the exception handling at this place.

**Comment by Belphemur** [comment]
Holy Shit ...

make sense ... it crashs at each "painting" of the progress bar ... Well 
clearly disable the painting if the theme is not supported.

**Comment by FireEmerald** [comment]
Is there anything special when compiling SoundSwitch for older WinVersions, like Win7? I only get a APPCRASH after starting the *.exe if i use my compiled version.

I thought it would be enought to install the [Visual C++ 2015 Build Tools](http://landinghub.visualstudio.com/visual-cpp-build-tools).

**Comment by Belphemur** [comment]
Use the Make.bat to build the project. It will generate everything correctly.

It's easier to use the installer on the VM to try it out.

**Comment by FireEmerald** [comment]
Fixed in aa48697f2fc2b9b4c65408128750d500480b8dbe.

**Comment by Belphemur** [comment]
Can you do a pull request on dev ?

In the message put "Fixes #194", I'll merge it :)
